File: ros2/py_topic/py_topic/publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String


class MyNode(Node):
    def __init__(self):
        super().__init__("py_talker_node")
        self.counter_ = 0
        self.publishers_ = self.create_publisher(String, "py_test_counter", 10)
        self.timer_ = self.create_timer(0.5, self.publisherCallback)
        self.get_logger().info("py_talker started...")

    def publisherCallback(self):
        self.counter_ += 1
        msg = String()
        msg.data = str(self.counter_)+") Hello"
        self.publishers_.publish(msg)
        self.get_logger().info(str(self.counter_)+") Hello")


def main(args=None):
    rclpy.init(args=args)
    node = MyNode()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: ros2/cpp_param/src/publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"
#define FREQ 10

// using namespace std::chrono_literals  // 1s gibi kavramların sistemde doğrudan kullanılmasını sağlarız
using string_msg = example_interfaces::msg::String;

class MyNode: public rclcpp::Node{
public:
    MyNode():Node("cpp_talker_node"){
        this->declare_parameter("cpp_class_name",2);
        this->declare_parameter("freq_pub", FREQ);
        frequency = this->get_parameter("freq_pub").as_int();
        rclcpp::Parameter number_ = this->get_parameter("cpp_class_name");
        RCLCPP_INFO(this->get_logger(), "%ld) Talker started... ", number_.as_int());
        publisher_ = this->create_publisher<string_msg>("cpp_test_counter", 10);
        timer_ = this->create_wall_timer(std::chrono::milliseconds((int)(1000.0 /frequency)), std::bind(&MyNode::publisherCallback, this));
    }
    
private:
    void publisherCallback(){
        RCLCPP_INFO(this->get_logger()," Hello");
        string_msg msg;  // auto msg = string_msg;
        msg.data = std::string(" Hello ");
        publisher_->publish(msg);
    }
    // int number_;
    int frequency;
    rclcpp::TimerBase::SharedPtr timer_;
    rclcpp::Publisher<string_msg>::SharedPtr publisher_;
    
};


int main(int argc, char ** args){
    rclcpp::init(argc, args);
    rclcpp::spin(std::make_shared<MyNode>());
    rclcpp::shutdown();
}